database/factories/DatasetFactory.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Database\Factories;
+
+use App\Models\Dataset;
+use Carbon\Carbon;
+use Illuminate\Database\Eloquent\Factories\Factory;
+use Illuminate\Support\Str;
+
+class DatasetFactory extends Factory
+{
+    /**
+     * The name of the factory's corresponding model.
+     *
+     * @var string
+     */
+    protected $model = Dataset::class;
+
+    /**
+     * Define the model's default state.
+     *
+     * @return array
+     */
+    public function definition()
+    {
+        $name = $this->faker->word().'dx';
```

Comment:
Why .dx ?? @NRayya

Reply:
jcamp datasets are named like this, but it doesn't really needed